### ISSUE TITLE
Add extra project advanced config option for new point collection: point, polygon, polyline

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#1979]
**Is your feature request related to a problem? Please describe.**

- We can currently add a new point in FMTM and have it injected as the `new_feature` field in the XLSForm.
- The XLSForm field is currently `geopoint`.
- Ideally we can have this configurable, so we inject the field as geotrace (polygon) or geoshape (polygon), and allow the user to draw the relevant geometry in FMTM to inject.

This is a follow up issue from https://github.com/hotosm/fmtm/issues/1915

**Describe the solution you'd like**

- [x] Add a config option for this during project creation (frontend, the backend accepts the param)
- [x] Add a db field for `projects` to capture this, as an enum with three options **edit: done, added field new_geom_type with enum options `POINT`, `POLYLINE`, `POLYGON`.
- [x] Based on the enum value in the project JSON in the frontend, we change the terradraw geom type: point, polyline, polygon.
- [x] We may need to add buttons for deleting and modifying vertices if drawing more complex polygons.
- [x] The `append_mandatory_fields` function in osm-fieldwork should have the option to specify the new geom type, so the XLSForm has the relevant field type. https://github.com/hotosm/osm-fieldwork/pull/329 (released in v0.18.0)

**Additional context**

- This may be dependent on moving the mandatory XLSForm questions sheet out of the XLS file and directly into code (to make things easier).

**Comment by manjitapandey** [comment]
> We can currently add a new point in FMTM and have it injected as the new_feature field in the XLSForm.

Is this new_feature becomes an entity and can we use it in visualising the bad submissions?? 

cc: @spwoodcock , @Sujanadh

**Comment by spwoodcock** [comment]
Yes we can for sure!

Either:

- Make an API call to get entity ids and statuses. Then fetch the geoms via their entity id.

- (preferred) insert the new geoms and bad geoms into an FMTM database table. Have the table reactive via electric again. The geometries will be loaded on the map real time as they are mapped or marked bad.

**Comment by spwoodcock** [comment]
I'm working on it - ticking off the tasks in the description as they are done

**Comment by manjitapandey** [comment]
@spwoodcock Does this needs frontend side of action now?? I am not sure as i am not being able to draw new polygon on FMTM mapper's page.
cc: @NSUWAL123

**Comment by spwoodcock** [comment]
> @spwoodcock Does this needs frontend side of action now?? I am not sure as i am not being able to draw new polygon on FMTM mapper's page.
> cc: @NSUWAL123 

If you create a new project, it will default to polygons.

Existing projects are point projects.

We don't have a way to specify this during project creation, but its not a problem for now, as I imagine most want polygons (can address that later)

**Comment by manjitapandey** [comment]
Thanks, this clarification helps me in testing.

**Comment by manjitapandey** [comment]
I am not finding the current frontend part intuitive. When I click on map new feature and click on map to draw, I am not confident on number of clicks i need to redirect to odk for form submissions. Sometimes, I am redirected to odk when only two nodes are drawn and sometimes I am not able to go to odk even after feature drawing is completed, rather it keeps editing the feature. 

cc: @NSUWAL123 can you verify the behavior??

**Comment by spwoodcock** [comment]
Hmm yes its very flakey on mobile!

There is a lot of interference from existing click events 

Perhaps when we are in draw mode, we can disable the click event for the rest of the map temporarily? (might be tricky)

Or another approach could be opening a popup map for drawing theb geom specifically (not ideal, but we could remove all the other click handlers in this case)

**Comment by spwoodcock** [comment]
The intended functionality can be tested on mobile here: https://terradraw.io/

You click to click an existing vertice for drawing to stop, which isn't super intuitive.

Perhaps we also need a button in place for the user to 'Confirm' the geometry and stop the drawing

**Comment by manjitapandey** [comment]
Yeah may be that work but I am unexpectedly redirected to odk even when I don't click on already existing node. We definitely have to switch to more intuitive drawing functionality help in UX drastically.

**Comment by spwoodcock** [comment]
Can the conflicting click events be looked into?

In the other map click events can we check if drawing is enabled, if so then prevent the default behaviour and return.

Maybe that will fix it

**Comment by NSUWAL123** [comment]
@spwoodcock I'll also look into this issue.

**Comment by spwoodcock** [comment]
> @spwoodcock I'll also look into this issue. 

Thank you!!

**Comment by manjitapandey** [comment]
The entity is not created for new geometries. When should the entities be created?? Is it when the user saves the new polygon geometry to get redirected to odk for form submission or is it possible to create entity once the submission is sent and also link the recent submission with just created entity?? 
@spwoodcock , @Sujanadh 
cc: @NSUWAL123

**Comment by spwoodcock** [comment]
> The entity is not created for new geometries. When should the entities be created?? Is it when the user saves the new polygon geometry to get redirected to odk for form submission or is it possible to create entity once the submission is sent and also link the recent submission with just created entity?? 
> @spwoodcock , @Sujanadh 
> cc: @NSUWAL123 

Entities in ODK should be created on form submission 👍 it needs to be looked into

**Comment by spwoodcock** [comment]
@Sujanadh suggested removing the entity creation logic from the form, instead creating the entity in FMTM directly.

See discussion here https://github.com/hotosm/osm-fieldwork/issues/338

This significantly simplfies our work, as XLSForms can be a bit finicky sometimes.

We also don't need a separate field to record new geoms this way, simply relying on selecting (or preselecting) an existing entity. We can then skip the geometry selection in ODK Collect entirely if its preselected (via intent URL)
